app/src/main/java/org/mifos/mobile/viewModels/UpdatePasswordViewModel.kt

```diff
@@ -0,0 +1,60 @@
+package org.mifos.mobile.viewModels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.observers.DisposableObserver
+import io.reactivex.schedulers.Schedulers
+import okhttp3.Credentials
+import okhttp3.ResponseBody
+import org.mifos.mobile.api.BaseApiManager
+import org.mifos.mobile.api.DataManager
+import org.mifos.mobile.api.local.PreferencesHelper
+import org.mifos.mobile.models.UpdatePasswordPayload
+import org.mifos.mobile.utils.MFErrorParser
+import javax.inject.Inject
+
+class UpdatePasswordViewModel @Inject constructor(
+    private val dataManager: DataManager?,
+    private val preferencesHelper: PreferencesHelper?
+) : ViewModel() {
+
+    private val compositeDisposable: CompositeDisposable = CompositeDisposable()
+    val passwordUpdateStatus: MutableLiveData<Boolean> = MutableLiveData()
+    val errorMessage: MutableLiveData<String> = MutableLiveData()
```

Comment:
Create a UI state class/interface, something similar to this -Don't copy past this code snippet I just wrote without using IDE lol

```
sealed interface UpdatePasswordUiState {
object Loading: UpdatePasswordUiState
object Success : UpdatePasswordUiState
data class Error(val error: String): UpdatePasswordUiState
}
```